pkg/utils/utils.go

```diff
@@ -37,3 +44,75 @@ func ParseInstanceID(node *v1.Node) (*string, error) {
 	}
 	return nil, fmt.Errorf("parsing instance id %s", node.Spec.ProviderID)
 }
+
+// GetSubnetFilters parses the AWSNodeTemplate to collect the subnet filters
+// from the Subnet tags
+func GetSubnetFilters(provider *v1alpha1.AWSNodeTemplate) []*ec2.Filter {
+	filters := []*ec2.Filter{}
+	// Filter by subnet
+	for key, value := range provider.Spec.SubnetSelector {
+		if key == "aws-ids" {
+			filterValues := functional.SplitCommaSeparatedString(value)
+			filters = append(filters, &ec2.Filter{
+				Name:   aws.String("subnet-id"),
+				Values: aws.StringSlice(filterValues),
+			})
+		} else if value == "*" {
+			filters = append(filters, &ec2.Filter{
+				Name:   aws.String("tag-key"),
+				Values: []*string{aws.String(key)},
+			})
+		} else {
+			filters = append(filters, &ec2.Filter{
+				Name:   aws.String(fmt.Sprintf("tag:%s", key)),
+				Values: []*string{aws.String(value)},
+			})
+		}
+	}
+	return filters
+}
+
+// SubnetIds parses the subnet object to collect the Subnet ids
+func SubnetIds(subnets []*ec2.Subnet) []string {
+	names := []string{}
+	for _, subnet := range subnets {
+		names = append(names, fmt.Sprintf("%s (%s)", aws.StringValue(subnet.SubnetId), aws.StringValue(subnet.AvailabilityZone)))
+	}
+	return names
+}
+
+// GetSubnetFilters parses the AWSNodeTemplate to collect the Security Group filters
+// from the Security Group tags
+func GetSecurityGroupFilters(provider *v1alpha1.AWSNodeTemplate) []*ec2.Filter {
```

Comment:
Typically, if at all possible, you want to avoid creating utils files and putting static helpers in them. These files tend to grow, and can become really unwieldy. Instead, if you can, it's best to encapsulate this logic with whatever area of the codebase uses it. In this case, it used to be encapsulated into the security group provider. Is there a reason you decided to move this?

Reply:
Removed from the utils files.